src/net/inv.rs

```diff
@@ -253,106 +408,426 @@ impl PeerBlocksInv {
     /// Count up the number of microblock streams represented
     pub fn num_microblock_streams(&self) -> u64 {
         let mut total = 0;
-        for i in self.first_block_height..self.num_sortitions {
+        for i in 0..self.num_sortitions {
             if self.has_ith_microblock_stream(i) {
                 total += 1;
             }
         }
         total
     }
+
+    /// Count up the number of anchor blocks represented
+    pub fn num_pox_anchor_blocks(&self) -> u64 {
+        let mut total = 0;
+        for i in 0..self.num_reward_cycles {
+            if self.has_ith_anchor_block(i) {
+                total += 1;
+            }
+        }
+        total
+    }
+    
+    /// Determine the lowest reward cycle that this pox inv disagrees with a given pox id
+    /// Returns (disagreed reward cycle, my-inv-bit, poxid-inv-bit)
+    /// If one is longer than the other, there will be disagreement
+    pub fn pox_inv_cmp(&self, pox_id: &PoxId) -> Option<(u64, bool, bool)> {
```

Comment:
This function has a corner case that I'm not sure matters (I think it indicates that one node has processed more Bitcoin blocks than the other node) -- when the lengths aren't equal, the shorter node will return "false" for the reward cycles it hasn't tried to process yet, such that there's cases where this function will return `Some(x, false, false)`, when what's really happening is just that one node is delayed from the other. Do you think this corner case could be problematic?

Reply:
Oh -- I see, you deal with this by just checking the prefixing in `pox_inv_prefix_cmp`

Reply:
That's the intended behavior.  If the caller gets `Some(x, b1, b2)`, the first thing the caller would do is compare `x` to its own PoX bit vector length and the other (`pox_id`) PoX vector in order to determine how to interpret `b1` and `b2`.  If `x` is equal to one of these vectors' lengths, then it would mean that `b1` and `b2` are not indicative of reward cycle disagreement _per se_ because in at least one PoX bit vector, the reward cycle at `x` is simply undefined.

In practice, only `x` is used by this method (the sister method `pox_inv_prefix_cmp` can be ignored, and will be removed, since no one is using it anymore).